mlir/lib/Tools/PDLL/Parser/Parser.cpp

```diff
@@ -1915,28 +1964,180 @@ FailureOr<ast::Expr *> Parser::parseEqualityExpr() {
 FailureOr<ast::Expr *> Parser::parseRelationExpr() { return parseAddSubExpr(); }
 
 FailureOr<ast::Expr *> Parser::parseAddSubExpr() {
-  auto lhs = parseMulDivExpr();
+  auto lhs = parseMulDivModExpr();
+  if (failed(lhs))
+    return failure();
+
+  for (;;) {
+    switch (curToken.getKind()) {
+    case Token::add: {
+      consumeToken();
+      auto rhs = parseMulDivModExpr();
+      if (failed(rhs))
+        return failure();
+      SmallVector<ast::Expr *> args{*lhs, *rhs};
+
+      // Check if it is in rewrite section but not in the let statement
+      bool inRewriteSection = parserContext == ParserContext::Rewrite;
+      if (inRewriteSection &&
+          nativeOperatorContext == NativeOperatorContext::Generic)
```

Comment:
Can't we check that this isn't a Let?
```suggestion
          nativeOperatorContext == NativeOperatorContext::Generic)
```